File: src-tauri/src/domain/response/role.rs
use crate::model::role::{RoleResponse, RoleResponseDeleteAt};

use super::PaginationMeta;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponseRoleAll {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponseRoleDelete {
    pub status: String,
    pub message: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponseRole {
    pub status: String,
    pub message: String,
    pub data: Option<RoleResponse>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponsesRole {
    pub status: String,
    pub message: String,
    pub data: Vec<RoleResponse>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponsePaginationRole {
    pub status: String,
    pub message: String,
    pub data: Vec<RoleResponse>,
    pub pagination: Option<PaginationMeta>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct ApiResponsePaginationRoleDeleteAt {
    pub status: String,
    pub message: String,
    pub data: Vec<RoleResponseDeleteAt>,
    pub pagination: Option<PaginationMeta>,
}

File: src/types/domain/response/role/role.ts
import { Role, RoleTrashed } from "@/types/model";
import { PaginationMeta } from "../pagination";

export interface ApiResponseRoleAll {
  status: string;
  message: string;
}

export interface ApiResponseRoleDelete {
  status: string;
  message: string;
}

export interface ApiResponseRole {
  status: string;
  message: string;
  data?: Role;
}

export interface ApiResponsesRole {
  status: string;
  message: string;
  data: Role[];
}

export interface ApiResponsePaginationRole {
  status: string;
  message: string;
  data: Role[];
  pagination: PaginationMeta;
}

export interface ApiResponsePaginationRoleDeleteAt {
  status: string;
  message: string;
  data: RoleTrashed[];
  pagination: PaginationMeta;
}
